core/src/main/java/io/parsingdata/metal/expression/value/NotAValue.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2013-2016 Netherlands Forensic Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.parsingdata.metal.expression.value;
+
+import java.math.BigInteger;
+import java.util.BitSet;
+import java.util.Objects;
+
+import io.parsingdata.metal.Util;
+import io.parsingdata.metal.data.Slice;
+import io.parsingdata.metal.encoding.Encoding;
+
+public final class NotAValue implements Value {
+
+    public static final Value NOT_A_VALUE = new NotAValue();
+    private static final UnsupportedOperationException UNSUPPORTED = new UnsupportedOperationException("NOT_A_VALUE does not support any Value operation.");
```

Comment:
Just wondering, because I've never "cached" an exception in a constant like this: what does the stack trace look like when this exception is thrown from one of the methods below?

Reply:
Good one, pretty sure the stack trace is filled at construction time. This behaviour could be overridden of course. But I would instead just add a method throwing this exception.

Reply:
The JLS indeed specifies that it is filled in when initialized. So I will fix this!